lex/outputbis.ml

```diff
@@ -29,54 +29,54 @@ type ctx = {
-    pr ctx "  else begin\n";
-    pr ctx "    lexbuf.Lexing.lex_curr_pos <- _curr;\n";
-    pr ctx "    lexbuf.Lexing.lex_last_pos <- _last;\n";
-    pr ctx "    __ocaml_lex_refill\n";
-    pr ctx "      (fun lexbuf ->\n";
-    pr ctx "        let _curr = lexbuf.Lexing.lex_curr_pos in\n";
-    pr ctx "        let _last = lexbuf.Lexing.lex_last_pos in\n";
-    pr ctx "        let _len = lexbuf.Lexing.lex_buffer_len in\n";
-    pr ctx "        let _buf = lexbuf.Lexing.lex_buffer in\n";
-    pr ctx "        if _curr < _len then\n";
-    pr ctx "          state lexbuf _last_action _buf _len (_curr + 1) \
-                            _last k\n";
-    pr ctx "            (Char.code (Bytes.unsafe_get _buf _curr))\n";
-    pr ctx "        else\n";
-    pr ctx "          __ocaml_lex_refill_buf lexbuf _buf _len _curr _last \
-                                             _last_action\n";
-    pr ctx "            state k\n";
-    pr ctx "      )\n";
-    pr ctx "      lexbuf\n";
-    pr ctx "  end\n";
-    pr ctx "\n";
-  end else begin
-    pr ctx "\n";
-    pr ctx "let rec __ocaml_lex_refill_buf lexbuf _buf _len _curr _last =\n";
-    pr ctx "  if lexbuf.Lexing.lex_eof_reached then\n";
-    pr ctx "    256, _buf, _len, _curr, _last\n";
-    pr ctx "  else begin\n";
-    pr ctx "    lexbuf.Lexing.lex_curr_pos <- _curr;\n";
-    pr ctx "    lexbuf.Lexing.lex_last_pos <- _last;\n";
-    pr ctx "    lexbuf.Lexing.refill_buff lexbuf;\n";
-    pr ctx "    let _curr = lexbuf.Lexing.lex_curr_pos in\n";
-    pr ctx "    let _last = lexbuf.Lexing.lex_last_pos in\n";
-    pr ctx "    let _len = lexbuf.Lexing.lex_buffer_len in\n";
-    pr ctx "    let _buf = lexbuf.Lexing.lex_buffer in\n";
-    pr ctx "    if _curr < _len then\n";
-    pr ctx "      Char.code (Bytes.unsafe_get _buf _curr), _buf, _len, \
-                            (_curr + 1), _last\n";
-    pr ctx "    else\n";
-    pr ctx "      __ocaml_lex_refill_buf lexbuf _buf _len _curr _last\n";
-    pr ctx "  end\n";
-    pr ctx "\n";
+  if ctx.has_refill then
+    pr ctx
+{|
+let rec __ocaml_lex_refill_buf lexbuf _buf _len _curr _last
+                                           _last_action state k =
+  if lexbuf.Lexing.lex_eof_reached then
+    state lexbuf _last_action _buf _len _curr _last k 256
+  else begin
+    lexbuf.Lexing.lex_curr_pos <- _curr;
+    lexbuf.Lexing.lex_last_pos <- _last;
+    __ocaml_lex_refill
+      (fun lexbuf ->
+        let _curr = lexbuf.Lexing.lex_curr_pos in
+        let _last = lexbuf.Lexing.lex_last_pos in
+        let _len = lexbuf.Lexing.lex_buffer_len in
+        let _buf = lexbuf.Lexing.lex_buffer in
+        if _curr < _len then
+          state lexbuf _last_action _buf _len (_curr + 1) _last k
+            (Char.code (Bytes.unsafe_get _buf _curr))
+        else
+          __ocaml_lex_refill_buf lexbuf _buf _len _curr _last
+                                             _last_action
+            state k
+      )
+      lexbuf
+  end
+|}
```

Comment:
I think that a newline is missing here -- compared to the previous output. Leaving a blank line at the end looks intentional.